plugins/xpressive/expressive_plugin.h

```diff
@@ -0,0 +1,254 @@
+const int	W2_EXPR = 1;
+const int	W3_EXPR = 2;
+const int	O1_EXPR = 3;
+const int	O2_EXPR = 4;
+const int	NUM_EXPRS = 7;
+
+
+class ExprFront;
+class WaveSample
+{
+public:
+	WaveSample(int _length)
+	{
+		length=_length;
+		samples=new float[length];
+		for(int i=0;i<length;++i)
+			samples[i]=0;
+	}
+	WaveSample(const graphModel * graph)
+	{
+		length=graph->length();
+		samples=new float[length];
+		memcpy(samples,graph->samples(),length*sizeof(float));
+	}
+	inline void copyFrom(const graphModel * graph)
+	{
+		memcpy(samples,graph->samples(),length*sizeof(float));
+	}
+	~WaveSample()
+	{
+		delete [] samples;
+	}
+	inline void setInterpolate(bool _interpolate) {interpolate=_interpolate;}
+	float * samples;
+	int length;
+	bool interpolate;
+};
+class exprSynth
+{
+
+	MM_OPERATORS
+public:
+	exprSynth(const WaveSample* gW1, const WaveSample* gW2, const WaveSample* gW3,ExprFront *_exprO1,ExprFront *_exprO2, NotePlayHandle * _nph,
+			const sample_rate_t _sample_rate,const FloatModel* _pan1,const FloatModel* _pan2,float _rel_trans);
+	virtual ~exprSynth();
+
+	void renderOutput( fpp_t _frames, sampleFrame * _buf );
+
+
+private:
+	ExprFront *exprO1, *exprO2;
+	const WaveSample *W1, *W2, *W3;
+	unsigned int note_sample;
+	unsigned int note_rel_sample;
+	float note_sample_sec;
+	float note_rel_sec;
+	float frequency;
+	float released;
+	NotePlayHandle* nph;
+	const sample_rate_t sample_rate;
+	const FloatModel *pan1,*pan2;
+	float rel_transition;
+	float rel_inc;
+
+} ;
+
+class expressive : public Instrument
+{
+	Q_OBJECT
+public:
+	expressive(InstrumentTrack * _instrument_track );
+	virtual ~expressive();
+
+	virtual void playNote( NotePlayHandle * _n,
+						sampleFrame * _working_buffer );
+	virtual void deleteNotePluginData( NotePlayHandle * _n );
+
+
+	virtual void saveSettings( QDomDocument & _doc,
+							QDomElement & _parent );
+	virtual void loadSettings( const QDomElement & _this );
+
+	virtual QString nodeName() const;
+
+	virtual f_cnt_t desiredReleaseFrames() const
+	{
+		return( 64 );
+	}
+
+	virtual PluginView * instantiateView( QWidget * _parent );
+
+protected:
+	static void smooth(float smoothness,const graphModel * in,graphModel * out);
+protected slots:
+
+
+private:
+	graphModel  m_graphO1;
+	graphModel  m_graphO2;
+	graphModel  m_graphW1;
+	graphModel  m_graphW2;
+	graphModel  m_graphW3;
+	graphModel  m_rawgraphW1;
+	graphModel  m_rawgraphW2;
+	graphModel  m_rawgraphW3;
+	IntModel m_selectedGraph;
+	QByteArray m_wavesExpression[3];
+	QByteArray m_outputExpression[2];
+	FloatModel m_parameterA1;
+	FloatModel m_parameterA2;
+	FloatModel m_parameterA3;
+	FloatModel m_smoothW1;
+	FloatModel m_smoothW2;
+	FloatModel m_smoothW3;
+	BoolModel m_interpolateW1;
+	BoolModel m_interpolateW2;
+	BoolModel m_interpolateW3;
+	FloatModel m_panning1;
+	FloatModel m_panning2;
+	FloatModel m_relTransition;
+	float m_A1,m_A2,m_A3;
+	WaveSample m_W1, m_W2, m_W3;
+
+
+	
+	BoolModel m_exprValid;
+	
+	friend class expressiveView;
```

Comment:
If you have some spare time, please use inline getters and setters instead of `friend`.